homeassistant/components/denonavr/media_player.py

```diff
@@ -188,49 +238,55 @@ def device_info(self):
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self._receiver.name
 
     @property
     def state(self):
         """Return the state of the device."""
-        return self._state
+        if self._available is False:
+            return STATE_UNAVAILABLE
```

Comment:
Set the `def available` `@property` instead as Home Assistant will take care of setting `STATE_UNAVAILABLE` if it returns True

Reply:
done